lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3InputFile.java

```diff
@@ -111,11 +112,14 @@ private GetObjectRequest newGetObjectRequest()
                 .requestPayer(requestPayer)
                 .bucket(location.bucket())
                 .key(location.key())
-                .applyMutation(builder -> key.ifPresent(encryption -> {
-                    builder.sseCustomerKey(encoded(encryption));
-                    builder.sseCustomerAlgorithm(encryption.algorithm());
-                    builder.sseCustomerKeyMD5(md5Checksum(encryption));
-                }))
+                .applyMutation(builder -> {
+                    key.ifPresent(encryption -> {
+                        builder.sseCustomerKey(encoded(encryption));
+                        builder.sseCustomerAlgorithm(encryption.algorithm());
+                        builder.sseCustomerKeyMD5(md5Checksum(encryption));
+                    });
+                    addEncryptionSettings(builder, context.s3SseContext());
```

Comment:
if (context.s3SseContext.type) { 
   checkState(key.isEmpty(), "");
}

Reply:
Comment was ignored. Can you elaborate?

Reply:
I added check in the constructor